src/core/geometry/fuzzyfactory/FuzzyFactory3.js

```diff
@@ -48,27 +50,26 @@ FuzzyShape3Factory.prototype = {
     let newvertices = sourcepolygon.vertices.map(function (vertex) {
       return _this.getVertex(vertex)
     })
-    // two vertices that were originally very close may now have become
-    // truly identical (referring to the same Vertex object).
-    // Remove duplicate vertices:
-    // uses the tags to set vertices appart
+        // two vertices that were originally very close may now have become
+        // truly identical (referring to the same Vertex object).
+        // Remove duplicate vertices:
     let newverticesDedup = []
     if (newvertices.length > 0) {
-      let prevvertextag = newvertices[newvertices.length - 1].getTag()
+      let prevvertex = newvertices[newvertices.length - 1]
       newvertices.forEach(function (vertex) {
-        let vertextag = vertex.getTag()
-        if (vertextag !== prevvertextag) {
+        if (!vec3.equals(prevvertex, vertex)) {
           newverticesDedup.push(vertex)
         }
-        prevvertextag = vertextag
+        prevvertex = vertex
       })
     }
-    // If it's degenerate, remove all vertices:
+        // If it's degenerate, remove all vertices:
```

Comment:
May want a comment about why a degenerate polygon with no vertices is preferable to a degenerate polygon with 1 or 2 vertices. :)